samples/dnn/dasiamrpn_tracker.cpp

```diff
@@ -0,0 +1,460 @@
+{
+    float x;
+    float y;
+    float w;
+    float h;
+} Box;
+
+// Initial parameters of the model
+typedef struct trackerConfig
+{
+    float windowInfluence = 0.43;
+    float lr = 0.4;
+    int scale = 8;
+    bool swapRB = false;
+    int totalStride = 8;
+    float penaltyK = 0.055;
+    int exemplarSize = 127;
+    int instanceSize = 271;
+    float contextAmount = 0.5;
+    std::vector<float> ratios = {0.33, 0.5, 1, 2, 3};
+    int anchorNum = ratios.size();
+    Mat anchors;
+    Mat windows;
+    Scalar avgChans;
+    Size imgSize = {0,0};
+    Box targetBox = {0,0,0,0};
+    int scoreSize = (instanceSize-exemplarSize)/totalStride+1;
+
+    void update_scoreSize()
+    {
+        scoreSize = int((instanceSize - exemplarSize) / totalStride + 1);
+    }
+}trackerConfig;
+
+void softmax(const Mat &src, Mat & dst);
+void elementMax(Mat & src);
+Mat generateHanningWindow(const trackerConfig &trackState);
+Mat generateAnchors(trackerConfig &trackState);
+Mat getSubwindow(Mat &img, const Box &targetBox, float originalSize, Scalar avgChans);
+float trackerEval(Mat img, trackerConfig &trackState, Net &siamRPN);
+void trackerInit(Mat img, trackerConfig &trackState, Net &siamRPN, Net &siamKernelR1, Net &siamKernelCL1);
+
+template<typename T>
+T sizeCal(const T &w, const T &h)
+{
+    T pad = (w + h)*0.5;
+    T sz2 = (w + pad) * (h + pad);
+    return sqrt(sz2);
+}
+
+template<>
+Mat sizeCal(const Mat &w, const Mat &h)
+{
+    Mat pad = (w + h)*0.5;
+    Mat sz2 = (w + pad).mul((h + pad));
+
+    cv::sqrt(sz2, sz2);
+    return sz2;
+}
+
+int main(int argc, char **argv)
+{
+    // Parse command line arguments.
+    CommandLineParser parser(argc, argv, keys);
+
+   if (argc == 1 || parser.has("help"))
+   {
+       parser.printMessage();
+       return 0;
+   }
+
+    std::string input = parser.get<String>("input");
+    std::string net = parser.get<String>("net");
+    std::string kernel_cls1 = parser.get<String>("kernel_cls1");
+    std::string kernel_r1 = parser.get<String>("kernel_r1");
+
+    // Read nets.
+    Net siamRPN, siamKernelCL1, siamKernelR1;
+    try
+    {
+        siamRPN = readNet(net);
+        siamKernelCL1 = readNet(kernel_cls1);
+        siamKernelR1 = readNet(kernel_r1);
+    }
+    catch (const cv::Exception& ee)
+    {
+        std::cerr << "Exception: " << ee.what() << std::endl;
+        std::cout << "Can't load the network by using the flowing files:" << std::endl;
+        std::cout << "siamRPN : " << net << std::endl;
+        std::cout << "siamKernelCL1 : " << kernel_cls1 << std::endl;
+        std::cout << "siamKernelR1 : " << kernel_r1 << std::endl;
+        return 1;
+    }
+
+    const std::string preWinName = "DaSiamRPN";
+    namedWindow(preWinName, WINDOW_AUTOSIZE);
+
+    // Open a video file or an image file or a camera stream.
+    VideoCapture cap;
+    Mat image;
+
+    bool openSuccess = parser.has("input") ? cap.open(parser.get<String>("input")) : cap.open(0);
+    CV_Assert(openSuccess);
```

Comment:
It is better to print the message with video file name instead of just `CV_Assert()`

Reply:
Thanks, fixed.